Public/Src/Cache/ContentStore/Distributed/Utilities/BandwidthCheckedCopier.cs

```diff
@@ -16,25 +17,39 @@ namespace BuildXL.Cache.ContentStore.Distributed.Utilities
     /// <summary>
     /// File copier that times out copies based on their bandwidths.
     /// </summary>
-    public class BandwidthCheckedCopier<T> : IFileCopier<T> where T : PathBase
+    public class BandwidthCheckedCopier : ITraceableAbsolutePathFileCopier
     {
-        private readonly IFileCopier<T> _inner;
+        private readonly IAbsolutePathFileCopier _inner;
         private readonly BandwidthChecker _checker;
         private readonly ILogger _logger;
 
         /// <nodoc />
-        public BandwidthCheckedCopier(IFileCopier<T> inner, BandwidthChecker.Configuration config, ILogger logger)
+        public BandwidthCheckedCopier(IAbsolutePathFileCopier inner, BandwidthChecker.Configuration config, ILogger logger)
         {
             _inner = inner;
             _checker = new BandwidthChecker(config);
             _logger = logger;
         }
 
         /// <inheritdoc />
-        public Task<CopyFileResult> CopyToAsync(T sourcePath, Stream destinationStream, long expectedContentSize, CancellationToken cancellationToken)
+        public Task<CopyFileResult> CopyToAsync(AbsolutePath sourcePath, Stream destinationStream, long expectedContentSize, CancellationToken cancellationToken)
         {
             var context = new OperationContext(new Context(_logger), cancellationToken);
             return _checker.CheckBandwidthAtIntervalAsync(context, token => _inner.CopyToAsync(sourcePath, destinationStream, expectedContentSize, token), destinationStream);
         }
+
+        /// <inheritdoc />
+        public Task<FileExistenceResult> CheckFileExistsAsync(AbsolutePath path, TimeSpan timeout, CancellationToken cancellationToken) => _inner.CheckFileExistsAsync(path, timeout, cancellationToken);
+
+        /// <inheritdoc />
+        public Task<CopyFileResult> CopyToAsync(OperationContext context, AbsolutePath sourcePath, Stream destinationStream, long expectedContentSize)
+        {
+            if (_inner is ITraceableAbsolutePathFileCopier traceable)
+            {
+                return _checker.CheckBandwidthAtIntervalAsync(context, token => traceable.CopyToAsync(context, sourcePath, destinationStream, expectedContentSize), destinationStream);
+            }
+
+            return _checker.CheckBandwidthAtIntervalAsync(context, token => _inner.CopyToAsync(sourcePath, destinationStream, expectedContentSize, token), destinationStream);

```

Comment:
Maybe just simplify this and call the extension method on _inner inside the lambda.